scripts/bootloader/zb_ota_sign.py

```diff
@@ -0,0 +1,266 @@
+              help='The zigbee OTA maximum hw version of Zigbee OTA Client.',
+              required=False,
+              type=BASED_INT_OR_NONE)
+def generate(zipfile,
+           application,
+           application_version,
+           application_version_string,
+           hw_version,
+           external_app,
+           zigbee,
+           zigbee_manufacturer_id,
+           zigbee_image_type,
+           zigbee_comment,
+           zigbee_ota_hw_version,
+           zigbee_ota_fw_version,
+           zigbee_ota_min_hw_version,
+           zigbee_ota_max_hw_version):
+    """
+    Generate a zip package for distribution to apps that support Nordic DFU OTA.
+    The application, bootloader, and SoftDevice files are converted to .bin if supplied as .hex files.
+    For more information on the generated package, see:
+    http://developer.nordicsemi.com/nRF5_SDK/doc/
+
+    The following combinations are supported by this command:
+
+    * APP only: Supported (external or internal).
+
+    """
+    zipfile_path = zipfile
+
+    # The user can specify the application version with two different
+    # formats. As an integer, e.g. 102130, or as a string
+    # "10.21.30". Internally we convert to integer.
+    if application_version_string:
+        application_version_internal = convert_version_string_to_int(application_version_string)
+        if application_version:
+            click.echo('Warning: When both application-version-string and application-version are provided, only the string will be used.')
+    else:
+        application_version_internal = application_version
+
+    if application_version_internal == 'none':
+        application_version_internal = None
+
+    if hw_version == 'none':
+        hw_version = None
+
+    if external_app is None:
+        external_app = False
+
+    if zigbee_ota_hw_version == 'none':
+        zigbee_ota_hw_version = None
+
+    if zigbee_ota_fw_version == 'none':
+        zigbee_ota_fw_version = None
+
+    if application is not None and application_version_internal is None:
+        raise click.UsageError('--application-version or --application-version-string'
+                   ' required with application image.')
+
+    if zigbee and zigbee_ota_hw_version is None:
+        raise click.UsageError("--zigbee-ota-hw-version is required.")
+
+    if zigbee and zigbee_ota_fw_version is None:
+        zigbee_ota_fw_version = 0
+
+    if zigbee_comment is None:
+        zigbee_comment = ''
+    elif any(ord(char) > 127 for char in zigbee_comment): # Check if all the characters belong to the ASCII range
+        click.echo('Warning: Non-ASCII characters in the comment are not allowed. Discarding comment.')
+        zigbee_comment = ''
+    elif len(zigbee_comment) > 30:
+        click.echo('Warning: truncating the comment to 30 bytes.')
+        zigbee_comment = zigbee_comment[:30]
+
+    if zigbee_manufacturer_id is None:
+        zigbee_manufacturer_id = 0xFFFF
+
+    if zigbee_image_type is None:
```

Comment:
What about introducing some Enum to define very basic img_types, instead of using wildcards? For example, `zigbee_app` can be defined at first.